presto-main/src/main/java/com/facebook/presto/sql/gen/VarArgsToArrayAdapterGenerator.java

```diff
@@ -50,116 +48,68 @@ private VarArgsToArrayAdapterGenerator()
     {
     }
 
-    public static class MethodHandleAndConstructor
+    public static class VarArgMethodHandle
     {
         private final MethodHandle methodHandle;
-        private final MethodHandle constructor;
 
-        private MethodHandleAndConstructor(MethodHandle methodHandle, MethodHandle constructor)
+        private VarArgMethodHandle(MethodHandle methodHandle)
         {
             this.methodHandle = requireNonNull(methodHandle, "methodHandle is null");
-            this.constructor = requireNonNull(constructor, "constructor is null");
         }
 
         public MethodHandle getMethodHandle()
         {
             return methodHandle;
         }
-
-        public MethodHandle getConstructor()
-        {
-            return constructor;
-        }
     }
 
-    @UsedByGeneratedCode
-    public static final class VarArgsToArrayAdapterState
-    {
-        /**
-         * User state created by provided user state factory
-         */
-        public final Object userState;
-
-        /**
-         * Array of argument, such as long[], Block[]
-         */
-        public final Object args;
-
-        public VarArgsToArrayAdapterState(Object userState, Object args)
-        {
-            this.userState = userState;
-            this.args = requireNonNull(args, "args is null");
-        }
-    }
-
-    public static MethodHandleAndConstructor generateVarArgsToArrayAdapter(
+    public static VarArgMethodHandle generateVarArgsToArrayAdapter(
             Class<?> returnType,
             Class<?> javaType,
             int argsLength,
-            MethodHandle function,
-            MethodHandle userStateFactory)
+            MethodHandle function)
     {
         requireNonNull(returnType, "returnType is null");
         requireNonNull(javaType, "javaType is null");
         requireNonNull(function, "function is null");
-        requireNonNull(userStateFactory, "userStateFactory is null");
 
-        checkCondition(argsLength <= 253, NOT_SUPPORTED, "Too many arguments for vararg function");
+        checkCondition(argsLength <= 254, NOT_SUPPORTED, "Too many arguments for vararg function");
```

Comment:
What is this change about?

Reply:
We removed state so we can allow one more param!